components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java

```diff
@@ -153,21 +161,24 @@ public void validate() {
     @Override
     public void configureForFilterOutcome(AnalysisJobBuilder ajb, FilterDescriptor<?, ?> descriptor, String categoryName) {
         final String dsName = ajb.getDatastore().getName();
-        sheetName = fixSheetName("output-" + dsName + "-" + descriptor.getDisplayName() + "-" + categoryName);
+        sheetName = fixSheetName(dsName + "-" + descriptor.getDisplayName() + "-" + categoryName);
     }
 
     @Override
     public void configureForTransformedData(AnalysisJobBuilder ajb, TransformerDescriptor<?> descriptor) {
         final String dsName = ajb.getDatastore().getName();
-        sheetName = fixSheetName("output-" + dsName + "-" + descriptor.getDisplayName());
+        sheetName = fixSheetName(dsName + "-" + descriptor.getDisplayName());
     }
 
     private String fixSheetName(String sheet) {
-        for (char c : ILLEGAL_SHEET_CHARS) {
+        for (char c : SHEET_NAME_ILLEGAL_CHARS) {
             while (sheet.indexOf(c) != -1) {
                 sheet = sheet.replace(c, '-');
             }
         }
+        if (sheet.length() > SHEET_NAME_MAX_LENGTH) {
```

Comment:
I realize that I should probably elaborate: Beyond removing "output-", which I guess is still a small part, I was thinking something along putting the datastore name into the file name (that would also improve the actual filename, right now it is the horribly generic "_DataCleaner-staging.xlsx_"), and only put the transformer name into the sheet name.